source/plugins/ruby/out_mdm.rb

```diff
@@ -408,5 +422,14 @@ def isWindows()
       end
       return isWindows
     end
+
+    def valid_url?(string)
+      uri = URI.parse(string)
```

Comment:
URI.parse should take care of throwing an exception if the URI is not valid. why do we need this helper method which does same thing what URI.parse does?

Reply:
Yep. URI.parse would raise [URI::InvalidURIError](https://apidock.com/ruby/URI/InvalidURIError) if it's not valid. Considering there's exception rescue in start function. Have replaced this helper method with URI.parse directly.